### PR TITLE
Make a map upload

### LINKED ISSUE #105: Make a map upload option

### PR DESCRIPTION
**Pull request by axel1200**
Adds a map upload pop up in the "tools" drop-down-menu
Fixes #105

### CODE REVIEW

**Comment by axel1200** [comment]
@ozonexo3 I would like you to acctually check the files that were edited are those I wanna push and not some files my IDE changed/needs and that should not be on GitHub. I am a total beginner in Unity so maybe you looking over it would be good to make sure I did not run into any stupid beginner issues. Maybe if you built a new version which you are probably gonna do, at least I assume. You could upload a test map. Of course I tired from within Unity but another test would definitely help to assure it works.

File: Assets/Scripts/UI/Windows/UploadMapPopUp.cs [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    public class UploadMapPopUp : MonoBehaviour
+    {
+        public InputField BaseUrlInputField;
+        public InputField MapFolderPathInputField;
+        public Toggle RankedToggle;
+        public Text ErrorText;
+        public InputField UsernameInputField;
+        public InputField PasswordInputField;
+
+        public void setMapFolderClicked()
```
Comment (Rackover) [inline Assets/Scripts/UI/Windows/UploadMapPopUp.cs]: PascalCase

File: Assets/Scripts/Common/ZipUtil.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+namespace Ozone
+{
+    public class ZipUtil
+    {
+
+
+		// Compresses the files in the nominated folder, and creates a zip file on disk named as outPathname.
+		//
+		public static void createZipFromFolder(string outPathname, string password, string folderName) {
```
Comment (Rackover) [inline Assets/Scripts/Common/ZipUtil.cs]: PascalCase

File: Assets/Scripts/Common/ZipUtil.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+				}
+				zipStream.CloseEntry();
+			}
+			string[ ] folders = Directory.GetDirectories(path);
+			foreach (string folder in folders) {
+				CompressFolder(folder, zipStream, folderOffset);
+			}
+		}
+    }
+}
```
Comment (Rackover) [inline Assets/Scripts/Common/ZipUtil.cs]: Line break

File: Assets/Scripts/UI/Windows/UploadMapPopUp.cs [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+            {
+                TokenEndpoint = new Uri(BaseUrlInputField.text+"/oauth/token"),
+                ProtocolVersion = ProtocolVersion.V20
+            };
+
+            var client = new UserAgentClient(server)
+            {
+                //Those are the credentials for Downlords-FAF-Client
+                ClientIdentifier = "3bc8282c-7730-11e5-8bcf-feff819cdc9f",
+                ClientCredentialApplicator = ClientCredentialApplicator.PostParameter("6035bd78-7730-11e5-8bcf-feff819cdc9f")
```
Comment (Rackover) [inline Assets/Scripts/UI/Windows/UploadMapPopUp.cs]: Should not be hardcoded
And even more, should **not** use the Downlord FAF Client credentials but rather be registered as a new oauth app

File: Assets/Scripts/UI/Windows/UploadMapPopUp.cs [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+                UsernameInputField.text, PasswordInputField.text);
+	
+            return token;
+        }
+        
+        //Copied from StackOverFlow
+        private static void DirectoryCopyWithSourceFolder(string sourceDirName, string destDirName, bool copySubDirs)
+        {
+            var newDest = Path.Combine(destDirName, Path.GetFileName(sourceDirName));
+            Debug.Log(newDest);
```
Comment (Rackover) [inline Assets/Scripts/UI/Windows/UploadMapPopUp.cs]: Extra Debug.Log with no comments nor precisions

File: Assets/Scripts/UI/Windows/UploadMapPopUp.cs [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+            {
+                onError("Sending failed. Maybe map is too big... Exception was: \n"+e.Message);
+            }
+
+            httpClient.Dispose();
+            form.Dispose();
+            GenericInfoPopup.ShowInfo("Uploaded Successful... Check the vault to see ;)");
+        }
+
+        private bool checkUserInput()
```
Comment (Rackover) [inline Assets/Scripts/UI/Windows/UploadMapPopUp.cs]: PascalCase

File: Assets/Scripts/UI/Windows/UploadMapPopUp.cs [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+            if (string.IsNullOrEmpty(PasswordInputField.text))
+            {
+                onError("Please enter a password");
+                return false;
+            }
+
+            return true;
+        }
+
+        private void onError(string errorMessage)
```
Comment (Rackover) [inline Assets/Scripts/UI/Windows/UploadMapPopUp.cs]: PascalCase

File: Assets/Scripts/UI/Windows/UploadMapPopUp.cs [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+            if (!Directory.Exists(destDirName))
+            {
+                Directory.CreateDirectory(destDirName);
+            }
+        
+            // Get the files in the directory and copy them to the new location.
+            FileInfo[] files = dir.GetFiles();
+            foreach (FileInfo file in files)
+            {
+                string temppath = Path.Combine(destDirName, file.Name);
```
Comment (Rackover) [inline Assets/Scripts/UI/Windows/UploadMapPopUp.cs]: camelCase for variables

**Comment by axel1200** [comment]
@Rackover thank u for your review 👍

File: Assets/Scripts/UI/Windows/UploadMapPopUp.cs (hunk 4 above)
Comment (axel1200) [reply Assets/Scripts/UI/Windows/UploadMapPopUp.cs]: there is no setting file so there is no other option then hard code

**Comment by axel1200** [comment]
@ozonexo3 your biggest fan if you have a look at it :D

File: Assets/Scripts/UI/Windows/UploadMapPopUp.cs (hunk 4 above)
Comment (Rackover) [reply Assets/Scripts/UI/Windows/UploadMapPopUp.cs]: Wrong, the other option is : make the setting file.
Don't mark this as resolved, this is *not* resolved.

**Comment by Rackover** [comment]
If the map editor has no settings / conf file where to write the oauth client id and secret, then make one - don't hardcode it
Also, stop abusing Downlord's client credentials for apps that are not Downlord's client. Ask a server admin to make credentials for that oauth app instead. You know at least 4 server admin personally.

**Comment by axel1200** [comment]
I did so i had Brutus make a new oauth client.But well what is a config file gonna give you ???? Whatever lets have @ozonexo3 say how he wants it. Also I do admit I got no clue how to make a config file in a unity project

**Pull request merged**